onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -58,18 +72,198 @@ bool HasExternalInitializer(const InitializedTensorSet& initializers, const Node
   return false;
 }
 
-template <class T>
-void CreateSharedOpSupportCheckerImpl(const std::string& op_type,
-                                      OpSupportCheckerRegistrations& op_registrations,
-                                      const std::vector<std::string>& op_types) {
-  // The shared OpSupportChecker is already in the OpSupportCheckerRegistrations
-  if (op_registrations.op_support_checker_map.find(op_type) != op_registrations.op_support_checker_map.cend())
-    return;
+static bool IsQuantizationScaleSupported(const InitializedTensorSet& initializers,
+                                         const NodeUnitIODef& io_def,
+                                         const OpSupportCheckParams& params,
+                                         const std::string& op_type,
+                                         bool is_quant_matmul,
+                                         bool is_conv_matmul_u8s8_weight) {
+  const auto scale_name = io_def.quant_param->scale.Name();
+  if (!Contains(initializers, scale_name)) {
+    LOGS_DEFAULT(VERBOSE) << "The scale of " << op_type << " must be an initializer tensor";
+    return false;
+  }
 
-  op_registrations.support_checkers.push_back(std::make_unique<T>());
-  for (const auto& op : op_types) {
-    op_registrations.op_support_checker_map.emplace(op, op_registrations.support_checkers.back().get());
+  const auto& scale_tensor = *initializers.at(scale_name);
```

Comment:
>at [](http://example.com/codeflow?start=43&length=2)

general comment: double lookup of `Contains(container, key)` then `container.at(key)` can be changed to `auto it = container.find(key); it != container.end()` and `it->second`

Reply:
also shows up in HasExternalInitializer(), IsQuantizationZeroPointSupported(), ReshapeOpSupportChecker::IsOpSupportedImpl(), and ConvOpSupportChecker::IsOpSupportedImpl()
can also address it later